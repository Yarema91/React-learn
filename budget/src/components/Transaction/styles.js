import styled from 'styled-components';
// import Transaction from '.';

export const Wrapper = styled.div`
     background-color: ${({ value }) => value < 0 ? '#fdfde1' : '#fff'};
     border: 1px solid #dbdbdb ;
     border-radius: 3px;
     padding: 5px;
     margin: 10px;
     `;

Wrapper.displayName = 'TransactionWrapper'

export const TransactionDate = styled.div`
     flex-grow: 1;
     `;
TransactionDate.displayName = 'TransactionDate'

export const Value = styled.div`
     flex-grow: 1;
     `;
Value.displayName = 'Value'

export const Comment = styled.div`
     flex-grow: 2;
     `;
Comment.displayName = 'Comment'
 
