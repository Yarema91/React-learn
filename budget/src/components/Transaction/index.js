import PropTypes from 'prop-types';
import {Wrapper,TransactionDate, Value, Comment } from './styles';
// import Transactions from '../Transactions';

const Transaction = ({transaction: {value, date, comment}}) => (
    <Wrapper value={value}> 
        <TransactionDate>{date}</TransactionDate>
        <Value>{value.toFixed(2)}</Value>
        <Comment>{comment}</Comment>
        <br/>
    </Wrapper>                            
); 

Transaction.propTypes = {
    transaction: PropTypes.shape({
        label: PropTypes.string,
        value: PropTypes.number
    })
}

Transaction.defould = {
    transactions: {
        label: '',
        value: 0
    }
    }

export default Transaction;