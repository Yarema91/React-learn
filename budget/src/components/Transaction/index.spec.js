import React from 'react';
import {shallow} from 'enzyme';
import Transaction from '.';


describe('Transaction components', () => {
    let sut;
    let props;
    
    beforeEach(() => {
         props = {
            transaction: {
                value: 23,
                comment: 'test',
                date: '01.01.21'
    
            }
        };
        sut = shallow( <Transaction {...props}/>);
    })
    it('should show transaction', () => { 
        expect(sut).toMatchSnapshot();
    });
    it('should show  two zeros after amount', () => { 
        expect(sut.find('Value').text()).toBe('23.00')
        
    });
})