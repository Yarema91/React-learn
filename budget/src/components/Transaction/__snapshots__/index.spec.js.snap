// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction components should show transaction 1`] = `
<TransactionWrapper
  value={23}
>
  <TransactionDate>
    01.01.21
  </TransactionDate>
  <Value>
    23.00
  </Value>
  <Comment>
    test
  </Comment>
  <br />
</TransactionWrapper>
`;
